test/geometry_tests.cpp

```diff
@@ -10,156 +10,241 @@
 // Code to test
 #include "gillwald/geometry.hpp"
 
-namespace geometry {
-
-void printPixels(std::vector<std::vector<int>> pixels){
-	std::cout << "\n";
-	for (const auto &pixel:pixels ) {
-		std::cout << "(" << pixel[0] << ", " << pixel[1] << ")\n";		
-	}		
-	std::cout << "\n";
-}
-
-bool allXOrYCoorSame(int x_or_y, std::vector<std::vector<int>> pixels){
-	if (x_or_y == 0) { // Check X coordinates
-		x_or_y = 0;
-	} else { // Check Y coordinatesd
-		x_or_y = 1;
-	}
-
-	auto shouldBeEqualTo = pixels[0][x_or_y];
-
-	float sum = 0;
+using ::testing::ElementsAre;
+using ::testing::ElementsAreArray;
 
-	for (const auto &pixel:pixels) {
-		sum += pixel[x_or_y];
-	}
-
-	float avg = sum/static_cast<float>(pixels.size());
+namespace geometry {
 
-	if (avg == shouldBeEqualTo) {
-		return true;
-	} else {
-		return false;
+std::ostream& operator<<(std::ostream& os, const std::vector<Cell>& v){
+	os << "\n";
+	for (const auto& pixel:v) {
+		os << "(" << pixel.x << "," << pixel.y << ")\n";
 	}
-
+	os << "\n";
+	return os;
 }
 
-bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe, std::vector<std::vector<int>> v) {
-	std::vector<std::vector<int>>::iterator ptr;
-	if (firstOrLastPoint == 0) {
-		ptr = v.begin();
-	} else {
-		ptr = v.end() - 1;
-	}
-
-	if (((*ptr)[0] == x_shouldbe) && ((*ptr)[1] == y_shouldbe)) {
-		return true;
-	} else {
-		return false;
-	}
-
+inline bool operator==(const Cell &lhs,const Cell &rhs) {
+	return (lhs.x == rhs.x) && (lhs.y == rhs.y);
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_test) {
```

Comment:
Still need to fix the test names for both the suite and specific test, otherwise when this test fails, the output is super meaningless

Reply:
`TEST(bresenham_conversion, front_and_back_cells)`

Reply:
Done